numpy/lib/format.py

```diff
@@ -166,14 +166,130 @@
 ARRAY_ALIGN = 64 # plausible values are powers of 2 between 16 and 4096
 BUFFER_SIZE = 2**18  # size of buffer for reading npz files in bytes
 
+
+class MagicVersion(object):
+
+    def __init__(self, magic_string):
+        self._magic_string = magic_string
+
+    def create_magic_version_string(self, major_version, minor_version):
+        if not (0 <= major_version < 256):
+            raise ValueError("major version must be 0 <= major < 256")
+        if not (0 <= minor_version < 256):
+            raise ValueError("minor version must be 0 <= minor < 256")
+        if sys.version_info[0] < 3:
+            return self._magic_string + chr(major_version) + chr(minor_version)
+        else:
+            return self._magic_string + bytes([major_version, minor_version])
+
+    def retrieve_version_from_magic_version_string(self, magic_version_string):
+        self._check_magic_version_string(magic_version_string)
+        if sys.version_info[0] < 3:
+            return ord(magic_version_string[-2]), ord(magic_version_string[-1])
+        else:
+            return magic_version_string[-2], magic_version_string[-1]
+
+    def _check_magic_version_string(self, magic_version_string):
+        if magic_version_string[:-2] != self._magic_string:
+            error_message = "the magic string is not correct; expected {}, got {}"
+            raise ValueError(error_message.format(self._magic_string, magic_version_string[:-2]))
+
+
+class HeaderVersion(object):
+
+    AUTOMATIC = None
+    VERSION_1 = (1, 0)
+    VERSION_2 = (2, 0)
```

Comment:
Seems redundant - `version = (2, 0)` seems clearer and more extensible that `version = VERSION_2` to me.